components/org.wso2.micro.integrator.core/src/main/java/org/wso2/micro/core/CarbonAxisConfigurator.java

```diff
@@ -238,6 +241,26 @@ public boolean isGlobalyEngaged(AxisModule axisModule) {
         return false;
     }
 
+    /**
+     * Set the hotDeployment variable after reading the hot deployment parameter from the axis configuration. If the
+     * value is null, then the hotDeployment variable will be set to false which is the default value.
+     */
+    private void setHotDeploymentFeature() {
+        Parameter hotDeployment = axisConfig.getParameter(org.wso2.micro.integrator.core.Constants.HOT_DEPLOYMENT);
+        if (hotDeployment != null) {
```

Comment:
As isTrue() method can handle null value and will return false, do we need this null check?

Reply:
Yes. `isTrue()` is evaluating `hotDeployment.getValue()` (not the `hotDeployment` value itself). The null check is there to avoid any NullPointerExceptions when getting `hotDeployment.getValue()`.